test/e2e/e2e_suite_test.go

```diff
@@ -58,3 +59,19 @@ func createTestCatalog(ctx context.Context, name string, imageRef string) (*cata
 	err := c.Create(ctx, catalog)
 	return catalog, err
 }
+
+func patchCatalogImageRef(ctx context.Context, name string, newImageRef string) (*catalogd.ClusterCatalog, error) {
+	// Fetch the existing ClusterCatalog
+	catalog := &catalogd.ClusterCatalog{}
+	err := c.Get(ctx, client.ObjectKey{Name: name}, catalog)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the ImageRef
+	catalog.Spec.Source.Image.Ref = newImageRef
+
+	// Patch the ClusterCatalog
+	err = c.Update(ctx, catalog)
+	return catalog, err
```

Comment:
This seems odd to not catch the error here, a caller of this would never know that the catalog they got back from this was the original or the updated one. IMO it would be better to catch the exception.

```suggestion
	err = c.Update(ctx, catalog)
	if err != nil {
		return nil, err
	}
	
	return catalog, err
```

Reply:
ack, thanks for catching that!